### ISSUE TITLE
Can't activeate oh-my-fish themes

### ISSUE THREAD

**Issue by ashkan-leo** [nix-community/home-manager#1429]
### Issue description

I'm using fish as my default shell on macOS. I can' activate the themes I install using oh-my-fish. The prompt doesn't get reloaded after adding one of the themes to my home-manager configuration file. I appreciate any help!

Following the example in source code, I set up the theme like this:

```nix
    plugins = with pkgs; [
      # https://github.com/jethrokuan/z
      {
        name = "z";
        src = fetchFromGitHub {
          owner = "jethrokuan";
          repo = "z";
          rev = "5f6c716b49ae24a5b36df18a6eceb7609d3ae81d";
          sha256 = "0kykhan9rdzy8anif5jp1iv3djrakhwk2arll3k93vaxm3np0gfm";
        };
      }

      # https://github.com/oh-my-fish/theme-bobthefish
      {
        name = "bobthefish";
        src = fetchFromGitHub {
          owner = "oh-my-fish";
          repo = "theme-bobthefish";
          rev= "a2ad38aa051aaed25ae3bd6129986e7f27d42d7b";
          sha256 = "1fssb5bqd2d7856gsylf93d28n3rw4rlqkhbg120j5ng27c7v7lq";
        };
      }

    ];

```

#### Maintainer CC
 @Jomik 
#### Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.5.0, macOS 10.15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixpkgs-20.09pre236005.2b7c0dcdaab"`
 - channels(ashkanaleali): `"darwin, home-manager, nixpkgs-20.09pre236343.78215a8395f, nixpkgs-unstable-20.09pre236343.78215a8395f"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by ashkan-leo** [comment]
Running `rg bobthefish` in `nix/store` shows the theme is installed somewhere, but I can't figure out how to make fish use the theme. I used to switch the themes using `omf` before.

**Comment by ashkan-leo** [comment]
I solved it by adding the following to my configuration file.

```nix
  xdg.configFile."fish/conf.d/plugin-bobthefish.fish".text = mkAfter ''
  for f in $plugin_dir/*.fish
    source $f
  end
  '';
```

It would be nice to mention in the document that these files that are installed by home-manager are not actually sourced and hence not used after installation. 

Alternatively, I'm willing to open a PR to add this as part of the package. I'd need some guidance on where this piece of code should go as I'm not fluent with nix -- I picked it up a few weeks ago.
